### PR TITLE
feat(Pipelines): Add connection picker

### PR DESCRIPTION
**Pull request by cheikhgwane**
Add support for connection picker.

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- Change 1
- Change 2

## How/what to test

A few words about what needs to be tested, along with specific testing instructions if needed.

## Screenshots / screencast

[Screencast from 11-08-2023 02:49:10 PM.webm](https://github.com/BLSQ/openhexa-frontend/assets/25453621/b19b03b9-0beb-4da7-bf14-7316f421d01c)

### CODE REVIEW

File: src/workspaces/features/RunPipelineDialog/ParameterField.tsx [hunk 1]
```diff
@@ -39,6 +42,18 @@ const ParameterField = (props: ParameterFieldProps) => {
       />
     );
   }
+
+  if (isConnectionParameter(parameter.type)) {
+    return (
+      <WorkspaceConnectionPicker
+        workspaceSlug={workspaceSlug || ""}
```
Comment (pvanliefland) [inline src/workspaces/features/RunPipelineDialog/ParameterField.tsx:49]: Can `workspaceSlug` be null or undefined?

File: src/workspaces/features/RunPipelineDialog/ParameterField.tsx [hunk 2]
```diff
@@ -39,6 +42,18 @@ const ParameterField = (props: ParameterFieldProps) => {
...
+        value={value ?? []}
```
Comment (pvanliefland) [inline src/workspaces/features/RunPipelineDialog/ParameterField.tsx:50]: Why do we have an empty array as default / fallback value?

**Review by pvanliefland — COMMENTED** [review]
@cheikhgwane nice job!

On my end, I have a couple of questions. Also, this PR could include a couple of tests...

File: src/workspaces/features/RunPipelineDialog/ParameterField.tsx (hunk 1 above)
Comment (cheikhgwane) [reply src/workspaces/features/RunPipelineDialog/ParameterField.tsx:49]: On the model yes.

File: src/workspaces/features/RunPipelineDialog/ParameterField.tsx (hunk 2 above)
Comment (cheikhgwane) [reply src/workspaces/features/RunPipelineDialog/ParameterField.tsx:50]: It's for preventing the component to go from  uncontrolled to controlled. https://legacy.reactjs.org/docs/uncontrolled-components.html

**Review by pvanliefland — APPROVED** [review]
LGTM 👍

**Pull request merged**